hackathon/models.py

```diff
@@ -0,0 +1,177 @@
+from django.db import models
+from django.utils import timezone
+
+from django.contrib.auth.models import User
+
+# Optional fields are ony set to deal with object deletion issues.
+# If this isn't a problem, they can all be changed to required fields.
+
+# The "updated" field isn't editable in admin. Neither is "submission" field.
+# A custom method to auto-fill created_by fields can be made
+
+# When it seemed relevant, I moved the ForeignKey field to another model and
+# used the one listed in the schema as the related_name. I may have been
+# erroneous to do so but it can be corrected easily.
+# My reasoning is explained in comments above the ForeignKey fields.
+
+
+# Create your models here.
+class Hackathon(models.Model):
+    """Model representing a Hackathon. It is connected by a foreign key to 
+    User, HackAwards and HackTeam. Optional Fields: judges, organiser.
+    "awards" and "teams" are related tables. They have been moved to
+    HackAwardCategory and HackTeam respectively. Please see comments there."""
+    created = models.DateTimeField(default=timezone.now)
```

Comment:
Yeah, I thought that as well. In the Django tutorial for Boutique Ado this is used because it is editable afterwards. I wasn't sure if it was necessary, so I'll swap it out.